### PR TITLE
add decision-making for snapshot ahead

### LINKED ISSUE #2025: [Enhancement]: 为ObAshBuffer、ObActiveSessHistTask增加提前落盘判断逻辑
**Enhancement**
ObAshBuffer、ObActiveSessHistTask内部维护read_pos、last_regular_snapshot_time，基于填充速率和期望速率判断是否需要提前落盘。

### PR DESCRIPTION
**Pull request by huasiy**
### Task Description
close #2025

### CODE REVIEW

File: src/share/ash/ob_active_sess_hist_task.cpp [hunk 1]
```diff
@@ -180,6 +183,14 @@ void ObActiveSessHistTask::runTimerTask()
         }
       }
     } // end for
+    elapsed_secs_++;
+    if (elapsed_secs_ % snapshot_interval == 0) {
+      elapsed_secs_ = 0;
+      if (do_snapshot_ahead()) {
+        // send rpc
```
Comment (rolandqi) [inline src/share/ash/ob_active_sess_hist_task.cpp:190]: should we update read pos after send rpc?

File: src/share/ash/ob_active_sess_hist_task.cpp [hunk 2]
```diff
@@ -32,6 +33,8 @@ using namespace oceanbase::sql;
 // a sample would be taken place up to 20ms after ash iteration begins.
 // if sample time is above this threshold, mean ash execution too slow
 constexpr int64_t ash_iteration_time = 40000;   // 40ms
+constexpr int snapshot_interval = 10;           // 10s
+constexpr int speed_threshold = 10;
```
[the remark this answers is not in the document]
Comment (huasiy) [reply src/share/ash/ob_active_sess_hist_task.cpp:39]: comments have been added

File: src/share/ash/ob_active_sess_hist_task.cpp (hunk 1 above)
Comment (huasiy) [reply src/share/ash/ob_active_sess_hist_task.cpp:192]: this pr is only for decision-making for snapshot ahead, internal state change will be handled in issue #2026 and #2027

**Review by rolandqi — APPROVED** [review]

**Comment by rolandqi** [comment]
LGTM

File: src/share/ash/ob_active_sess_hist_task.cpp (hunk 1 above)
Comment (rolandqi) [reply src/share/ash/ob_active_sess_hist_task.cpp:192]: TODO: read_pos_ modification
1. snapshot ahead
2. rpc broadcast
3. resize_ash_size

**Pull request merged**